pkg/controller/multiarchbuildconfig/multiarchbuildconfig.go

```diff
@@ -120,41 +113,43 @@ func (r *reconciler) reconcile(ctx context.Context, req reconcile.Request, logge
 	return nil
 }
 
-func (r *reconciler) handleMultiArchBuildConfig(ctx context.Context, mabc *v1.MultiArchBuildConfig) error {
+func (r *reconciler) handleMultiArchBuildConfig(ctx context.Context, mabc *v1.MultiArchBuildConfig) (resultErr error) {
+	var mutateFn func(mabcToMutate *v1.MultiArchBuildConfig)
```

Comment:
We only have 2 functionalities.  You can just update the mabc accordingly on each of the operations. No need to return a mutate function.

Reply:
What's the point of the function `UpdateMultiArchBuildConfig` if we mutate the mabc in place without using a function?

Reply:
UpdateMultiArchBuildConfig is more generic to avoid using multiple functions for different kinds of changes.

Reply:
If we modify the `mabc` without using a mutate function then `UpdateMultiArchBuildConfig` would not be needed anymore. If that's fine, I'm going to remove it.

Reply:
This is not what I am saying. You can run the `UpdateMultiArchBuildConfig` with the mutateFn you need. There is no point of return the mutateFn on each of the two operations. Instead you can mutate the object directly and take care of the error on a higher level.

Reply:
> Instead you can mutate the object directly

I was following the existing approach like in `checkAllBuildsSuccessful`, that returns the mutating function. Do you mean using `UpdateMultiArchBuildConfig` directly inside each of the functions `checkAllBuildsSuccessful`, `handlePushImageWithManifest` and `handleMirrorImage` ? If you meant so, then the whole approach would be:
```golang
func (r *reconciler) handleMultiArchBuildConfig() {
  // ...
  if c := getCondition(PushImageManifestDone); c != nil {
    if err := r.handlePushImageWithManifest(mabc); err != nil {
      return err
    }
  }
  // ...
}
```

Reply:
Yea. Something like that, but it would be better to check the last condition. And if all the builds are OK and there is no `PushImageManifestDone` condition, we run the manifest pusher. Let's just keep the logic simple.

Reply:
Checking the last condition won't be sufficient since after mirroring the image a `MirrorImageManifestDone` condition is added too (we're adding it because we want to follow the flow of an mabc). In that case the reconciler would mirror and update the mabc, then it will be triggered again and, if we just look at the last condition, we would end up using the `manifestPusher` again. That's why we need `getCondition()`.

Reply:
That's not entirely true. The only case where we want to run the mirroring is when the last condition is  `MirrorImageManifestDone`. So checking if the last condition makes absolute sense here.

Reply:
Why? Shouldn't we mirror only when the last condition is `PushImageManifestDone` as mirroring happens after we have assembled a manifest list? Then we are going to mirror and add `MirrorImageManifestDone`.

Reply:
Yea exactly. We should mirror when the last condition is `PushImageManifestDone`. There is no other case.